### PR TITLE
Make `VoxelArea:position()` return a vector with metatable

### LINKED ISSUE #12520: `VoxelArea:position()` doesn't return a vector with the metatable
https://github.com/minetest/minetest/blob/b15393c2a5e43e5493c767877cf48b1a510736d6/builtin/game/voxelarea.lua#L50-L65

### PR DESCRIPTION
**Pull request by AFCMS**
Make `VoxelArea:position()` return a vector with metatable (Fix #12520)

## How to test

```lua
local voxel = VoxelArea:new(10, 10)
assert(vector.check(voxel:position(1)))
```

### CODE REVIEW

File: builtin/game/voxelarea.lua [hunk 1]
```diff
@@ -48,20 +48,19 @@ function VoxelArea:indexp(p)
...
 
-	p.y = math.floor(i / self.ystride) + MinEdge.y
+	local y = math.floor(i / self.ystride) + MinEdge.y
 	i = i % self.ystride
 
-	p.x = math.floor(i) + MinEdge.x
+	local x = math.floor(i) + MinEdge.x
 
-	return p
+	return vector.new(x, y, z)
```
Comment (appgurueu) [inline builtin/game/voxelarea.lua:63]: Consider localizing `math.floor` and `vector.new` as this is a performance-critical function.

Comment (AFCMS) [reply builtin/game/voxelarea.lua:63]: Is this OK for readability?
```lua
local math_floor = math.floor
local vector_new = vector.new
```

Comment (appgurueu) [reply builtin/game/voxelarea.lua:63]: > Is this OK for readability?
> 
> ```lua
> local math_floor = math.floor
> local vector_new = vector.new
> ```

Yes, absolutely.

Comment (AFCMS) [reply builtin/game/voxelarea.lua]: Fixed

**Review by appgurueu — APPROVED** [review]

**Pull request merged**